layers/gpu/instrumentation/gpuav_instrumentation.cpp

```diff
@@ -177,45 +287,87 @@ void SetupShaderInstrumentationResources(Validator &gpuav, CommandBuffer &cb_sta
+                                                  gpuav.instrumentation_desc_set_bind_index_, 1, &instrumentation_desc_set,
+                                                  static_cast<uint32_t>(dynamic_offsets.size()), dynamic_offsets.data());
+                    break;
+                case PipelineLayoutSource::LastBoundDescriptorSet:
+                case PipelineLayoutSource::LastPushedConstants: {
+                    // Currently bound pipeline/set of shader objects may have bindings that are not compatible with last
+                    // bound descriptor sets: GPU-AV may create this incompatibility by adding its empty padding descriptor sets.
+                    // To alleviate that, since we could not get a pipeline layout from last pipeline binding (it was either
+                    // destroyed, or never has been created if using shader objects), a pipeline layout matching bindings of last
+                    // bound pipeline or
+                    // last bound shader objects is created and used.
+                    // If will also be cached: heuristic is next action command will likely need the same.
+
+                    const uint32_t bindings_counts_from_last_pipeline =
+                        GetBindingsCountFromLastBoundPipeline(gpuav, bind_point, cb_state, last_bound);
+
+                    // If the number of binding of the currently bound pipeline's layout (or the equivalent for shader objects) is
+                    // less that the number of bindings in the pipeline layout used to bind descriptor sets,
+                    // GPU-AV needs to create a temporary pipeline layout matching the the currently bound pipeline's layout
+                    // to bind the instrumentation descriptor set
+                    if (bindings_counts_from_last_pipeline <
+                        static_cast<uint32_t>(inst_binding_pipe_layout_state->set_layouts.size())) {
+                        VkPipelineLayout instrumentation_pipe_layout = CreateInstrumentationPipelineLayout(
+                            gpuav, bind_point, loc, last_bound, gpuav.dummy_desc_layout_,
+                            gpuav.GetInstrumentationDescriptorSetLayout(), gpuav.instrumentation_desc_set_bind_index_);
+
+                        if (instrumentation_pipe_layout != VK_NULL_HANDLE) {
+                            DispatchCmdBindDescriptorSets(cb_state.VkHandle(), bind_point, instrumentation_pipe_layout,
+                                                          gpuav.instrumentation_desc_set_bind_index_, 1, &instrumentation_desc_set,
+                                                          static_cast<uint32_t>(dynamic_offsets.size()), dynamic_offsets.data());
+                            DispatchDestroyPipelineLayout(gpuav.device, instrumentation_pipe_layout, nullptr);
+                        } else {
+                            return;
```

Comment:
need a comment or a `LogError` as if we are hitting this, something is wrong and blowing up aftewards probably

Reply:
Can add a comment, error already has been emitted in `CreateInstrumentationPipelineLayout` at this point. This return is just poor man error propagation ￣へ￣

Reply:
If someone report an issue, It would be nice to see the error message knowing it got here... hurts nothing to add it

```suggestion
                            gpuav.InternalError(cb_state.Handle(), loc, "Could not create instrumentation pipeline layout");
                            return;
```